### PR TITLE
Sharingcart issue: 37

### PR DESCRIPTION
**Pull request by DKSteffensen**
Several changes has been made to support new features and optimize older.

* General overhaul of UI to match moodle better.
* Corrected namespace naming to match moodle standard.
* Added feature to copy section from block, since some course formats did not work with sharing cart.
* Added modals instead of javascript confirms.
* New js is written ad amd module.
* Some functions in script.js has been moved to be more dynamic.
* Removed module.js.
* Testet in moodle versions: 3.6, 3.7 and 3.8.

https://github.com/donhinkelman/moodle-block_sharing_cart/issues/37

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -278,9 +360,9 @@ require(['jquery','core/str'], function ($, Str)
                     .parent().parent().find('h3.sectionname').text());
```
Comment (mcardle) [inline script.js]: This should probably be .closest('PARENT_SELECTOR').find(h3.sectionname').text());

File: script.js [hunk 2]
```diff
@@ -324,8 +407,8 @@ require(['jquery','core/str'], function ($, Str)
             }
 
             function open($dir, visible) {
-                var pix = icon[visible ? 'dir-open' : 'dir-closed'].pix;
-                $dir.find('> div img').first().attr('src', M.util.image_url(pix));
+                var iconElement = icon[visible ? 'dir-open' : 'dir-closed'].iconClass;
```
Comment (mcardle) [inline script.js]: It is bad practice to do logic inside an array - either do:

var iconElement = visible ? icon['dir-open'] : icon['dir-closed']

or:

var iconIndex = visible ? 'dir-open' : 'dir-closed';
var iconElement = icon[iconIndex]; // or just use it as icon[iconIndex] later on.

File: script.js [hunk 3]
```diff
@@ -403,10 +486,12 @@ require(['jquery','core/str'], function ($, Str)
                 this.hide();
 
                 function move(e) {
+
                     var m = $(e.target).closest('a').attr('class').match(/move-(\d+)-to-(\d+)/);
```
Comment (mcardle) [inline script.js]: Be more specific - what type of id's?

If I later on want to use a course id, could I use id or is it a section id or a cmid. Maybe call them section_id_from and section_id_to.

Comment (mcardle) [inline script.js]: Be more specific - what type of id's?

If I later on want to use a course id, could I use id or is it a section id or a cmid. Maybe call them section_id_from and section_id_to.

File: script.js [hunk 4]
```diff
@@ -1087,22 +1147,33 @@ require(['jquery','core/str'], function ($, Str)
             });
         };
 
+        $('');
```
Comment (mcardle) [inline script.js]: This should be removed

File: version.php [hunk 5]
```diff
@@ -24,8 +24,8 @@
 
 defined('MOODLE_INTERNAL') || die;
 
-$plugin->version   = 2020021200;
+$plugin->version   = 2020022701;
 $plugin->requires  = 2017050500; // Moodle 3.3
```
Comment (mcardle) [inline version.php]: This should only work for Moodle 3.6.

Comment (mcardle) [inline version.php]: This should only work for Moodle 3.6.

File: amd/src/bulkdelete.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+                    modal.show();
+                });
+            }
+
+            /**
+             *
+             * @returns {any[]}
+             */
+            function get_checks() {
+                var els = document.forms["form"].elements;
```
[the remark this answers is not in the document]
Comment (DKSteffensen) [reply amd/src/bulkdelete.js]: Rettet.

File: classes/section.php [hunk 7]
```diff
@@ -50,4 +49,10 @@ public static function get($course_id, $section_number) { global $DB;
 
 		return $section;
 	}
+
+	public static function all($course_id) {
+        get_fast_modinfo($course_id, 0, true);
```
[the remark this answers is not in the document]
Comment (DKSteffensen) [reply classes/section.php]: We talked about doing it this way, since the cache saved the information. Is it quicker to get the information from db?

**Pull request merged**